---
title: "COVID bulk sequencing project: sex differences"
author: "Helen Cai"
date: "2025-01-31"
output: word_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

The purpose of this notebook is to work through stratifying the data generated into M vs F, then re-performing analysis that has previously been done. To recap: the data were generated by Matt, and consist of brains dissected from pups born to mothers infected with COVID in a mild model. There are data from both E15 and E15. We have previously looked at this data and made a couple of slight adjustments. 

In this notebook, we will:

* read in metadata & separate samples by sex and age
* import the relevant salmon files
* perform standard limma analysis (including PCA and MDS plots)
* generate lists of DEGs
* create volcano plots and heatmaps for the DEGs

We will go forth with pathway analysis and getting lists of overlapping genes separately.

```{r, eval=T, echo=F, message=F}
library(ggplot2)
library(dplyr)
library(reshape)
library(MetBrewer)
library(tidyverse)
library(clusterProfiler)
library(tximport)
library(AnnotationHub)
library(limma)
library(edgeR)


library(AnnotationDbi)
library(org.Mm.eg.db)

library(pheatmap)
```

## Read in the metadata

```{r, eval=T, echo=F, message=F}

metadata <- read.csv("/home/hsc26/pi_pattabiraman/project/covid-bulk/covid_metadata_MY.csv", row.names=1)

table(metadata$age, metadata$condition)
table(metadata$age, metadata$sex)
attach(metadata)
```

## What files are we interested in?
This is where we'll need to do some tweaking to make sure we are slicing the right age and sex category. 

```{r, eval=T, echo=F, message=F, warning=F}
hub = AnnotationHub()
ensdb_query <- query(hub, c("EnsDb", "musculus", "100"))

# retrieve the record
ensdb_111 <- ensdb_query[["AH116340"]]
# convert to df
tx_data <- transcripts(ensdb_111, return.type="DataFrame")
tx2gene <- tx_data[, c("tx_id", "gene_id")]

# get paths of salmon output files
output_dir <- "/home/hsc26/pi_pattabiraman/project/covid-bulk/preprocess/salmon_outs/"
output_files <- paste0(output_dir, rownames(metadata), "/quant.sf")

names(output_files) <- rownames(metadata)
```

```{r, eval=T, echo=F, message=F}
######### change characteristics here ############
age_symb <- "P5"
sex_symb <- "F"

metadata_sex_group <- metadata[metadata$age == age_symb & sex == sex_symb,]
sex_group <- output_files[metadata$age == age_symb & sex == sex_symb]
```

## Read in Salmon files
```{r, eval=T, echo=F, message=F}
# nice tutorial for importing from txi here:
# https://bioconductor.org/packages/devel/bioc/vignettes/tximport/inst/doc/tximport.html#limma-voom

txi_brain <- tximport(sex_group, type="salmon", tx2gene=tx2gene,
                      countsFromAbundance="lengthScaledTPM", ignoreTxVersion=T)
```

## limma-voom analysis

```{r, eval=T, echo=F, message=F}

# create a DGEList()
y <- DGEList(txi_brain$counts)

# calculate normalization factors
y <- calcNormFactors(y)

# filter lowly-expressed genes
cutoff <- 1
drop <- which(apply(cpm(y), 1, max) < cutoff)
d <- y[-drop,]
dim(d)

# make sure that the metadata is in the same order as the column names of the counts table
colnames(y$counts) == rownames(metadata_sex_group)

```

```{r, eval=T, echo=F, message=F}

# open the plotting device
fname <- paste0("/home/hsc26/pi_pattabiraman/project/covid-bulk/sex-differences/", age_symb, "_", sex_symb, "/", age_symb, "_", sex_symb, "_", "MDS_plot.png")
png(filename = fname, width = 750, height = 450)

# make the multidimensional scaling (MDS) plot
colors <- as.numeric(as.factor(metadata_sex_group$condition))
symbols <- as.numeric(as.factor(metadata_sex_group$condition)) + 14
plot_title <- paste(age_symb, sex_symb, "MDS plot")
plotMDS(d, col=colors, pch=symbols, main=plot_title)
legend_labels <- levels(factor(metadata_sex_group$condition))
legend("bottomleft", legend=legend_labels, col=1:length(legend_labels),
       pch=(1:length(legend_labels))+14)

# close the plotting device
dev.off(0)
```

```{r, eval=T, echo=T, message=F}

# specify the model to be fitted
# we shouldn't have any interactions as they are all at the 
# same age and same sex
# thus, the only variable here is infection group
condition <- as.factor(metadata_sex_group$condition)
mm <- model.matrix(~0 + condition)

# voom!
y <- voom(d, mm, plot=T)
```

## PCA plot

```{r, eval=T, echo=F, message=F}
# recall that an EList holds expression values on the log scale 
# (e.g. after background correction and normalization)
pca <- prcomp(t(y$E))

pca_scores <- data.frame(pca$x)
pca_scores$Group <- condition

# create the ggplot
# Plotting the PCA results with ggplot2
plot_title <- paste0("Brain PCA plot (", age_symb, " ", sex_symb, ")")
if (sex_symb == "M") {shape <- 17} else shape <- 16


# add percent variance to the axes
pca_var <- pca$sdev^2
pca_var_per <- round(pca_var/sum(pca_var)*100, 1)
pc1_lab <- paste0("PC1 (", pca_var_per[1], "%)")
pc2_lab <- paste0("PC2 (", pca_var_per[2], "%)")


# create the ggplot
# Plotting the PCA results with ggplot2
ggplot(pca_scores, aes(x = PC1, y = PC2, color = Group)) +
  geom_point(size = 3, shape = shape) +  # Adjust size as needed
  labs(title = paste0("Brain PCA plot (", age_symb, " ", sex_symb, ")"), x = pc1_lab, y = pc2_lab) +
  scale_color_manual(values=c("#0086C3", "#FD4C4F")) +
  theme_classic(base_size=13) +
  theme(legend.position = "none")  # Position legend to the right


fname <- paste0("/home/hsc26/pi_pattabiraman/project/covid-bulk/sex-differences/", age_symb, "_", sex_symb, "/", age_symb, "_", sex_symb, "_", "PCA_plot.pdf")

ggsave(fname,
       height = 11,
       width = 9, 
       units="cm")


```

## Fit linear model

```{r, eval=T, echo=F, message=F}
# fit linear model using limma
# lmFit fits a linear model using weighted least squares for each gene
fit <- lmFit(y, mm)
coef.fit <- fit$coefficients
#head(coef(fit))


# comparisons between groups (log-fold changes) are obtained as contrasts of fitted linear models
# need to specify which groups to compare
# for instance, to compare SARS vs PBS:
contr <- makeContrasts(conditionsars - conditionpbs,
                       levels=colnames(coef(fit)))

# estimate contrast for each gene
tmp <- contrasts.fit(fit, contr)

# empirical Bayes smoothing of standard errors 
# (shrinks standard errors that are much larger or smaller than those from other genes towards the average standard error)
tmp <- eBayes(tmp)

# get DEGs
top_table <- topTable(tmp, sort.by="P", n=Inf)
```


```{r, echo=F, message=F, eval=T}
# convert into gene symbols
gene_db <- AnnotationDbi::select(org.Mm.eg.db, keys=rownames(top_table), keytype="ENSEMBL",
                  columns=c("SYMBOL"))

top_table_symbols <- merge(gene_db, top_table, by.x="ENSEMBL", by.y=0)
top_table_symbols <- top_table_symbols[sort(abs(top_table_symbols$logFC), decreasing = T, index.return=T)$ix,]
#head(top_table_sig_symbols, 15)
```

## Save the full DEG list
```{r, eval=F, echo=F, message=F}

fname <- paste0("/home/hsc26/pi_pattabiraman/project/covid-bulk/sex-differences/", age_symb, "_", sex_symb, "/", age_symb, "_", sex_symb, "_", "conditionsars-conditionpbs-FULL.csv")
write.csv(top_table_symbols %>% arrange(adj.P.Val), fname,
          row.names=F, quote=F)
```

## Volcano plot: logFC 0.5


```{r, eval=T, echo=F, message=F}
# Add a column for significance, usually set a threshold for adjusted p-value or logFC
top_table$Significant <- "Not Significant"
top_table$Significant[top_table$adj.P.Val < 0.05 & top_table$logFC > 0.5] <- "Significantly Upregulated"
top_table$Significant[top_table$adj.P.Val < 0.05 & top_table$logFC < -0.5] <- "Significantly Downregulated"
top_table$Significant[top_table$adj.P.Val < 0.05 & top_table$logFC > 0 & top_table$logFC < 0.5] <- "Moderately Upregulated"
top_table$Significant[top_table$adj.P.Val < 0.05 & top_table$logFC < 0 & top_table$logFC > -0.5] <- "Moderately Downregulated"


plot_title <- paste(age_symb, sex_symb)
# Create the volcano plot using ggplot2
ggplot(top_table, aes(x=logFC, y=-log10(adj.P.Val), color=Significant)) +
  geom_point(size=2) +
  scale_color_manual(values=c("Not Significant"="#BEBEBE", "Significantly Upregulated"="#EB4445",
                              "Significantly Downregulated"="#78B2D3", "Moderately Upregulated"="#F36F6E",
                              "Moderately Downregulated"="#A8CFE4")) +
  theme_classic(base_size=13) +
  labs(title="",
       x="",
       y=bquote(-log[10]~"P-adj"),
       color="Significance") +
  theme(legend.position="none") +
  geom_vline(xintercept=c(-0.5, 0.5), linetype="dashed", color="darkgray") +      
  geom_hline(yintercept=-log10(0.05), linetype="dashed", color="darkgray") 


fname <- paste0("/home/hsc26/pi_pattabiraman/project/covid-bulk/sex-differences/", age_symb, "_", sex_symb, "/", age_symb, "_", sex_symb, "_", "volcano-plot-logFC-05.pdf")
ggsave(fname,
       height = 11,
       width = 9,
       units = "cm")

```

```{r, eval=T, echo=F, message=F}
# Add a column for significance, usually set a threshold for adjusted p-value or logFC
top_table$Significant <- "Not Significant"
top_table$Significant[top_table$adj.P.Val < 0.05 & top_table$logFC > 1] <- "Significantly Upregulated"
top_table$Significant[top_table$adj.P.Val < 0.05 & top_table$logFC < -1] <- "Significantly Downregulated"
top_table$Significant[top_table$adj.P.Val < 0.05 & top_table$logFC > 0 & top_table$logFC < 1] <- "Moderately Upregulated"
top_table$Significant[top_table$adj.P.Val < 0.05 & top_table$logFC < 0 & top_table$logFC > -1] <- "Moderately Downregulated"


plot_title <- paste(age_symb, sex_symb)
# Create the volcano plot using ggplot2
ggplot(top_table, aes(x=logFC, y=-log10(adj.P.Val), color=Significant)) +
  geom_point(size=2) +
  scale_color_manual(values=c("Not Significant"="#BEBEBE", "Significantly Upregulated"="#EB4445",
                              "Significantly Downregulated"="#78B2D3", "Moderately Upregulated"="#F36F6E",
                              "Moderately Downregulated"="#A8CFE4")) +
  theme_classic(base_size=13) +
  labs(title="",
       x="Log2 Fold Change",
       y=bquote(-log[10]~"P-adj"),
       color="Significance") +
  theme(legend.position="none") +
  geom_vline(xintercept=c(-1,1), linetype="dashed", color="darkgray") +      
  geom_hline(yintercept=-log10(0.05), linetype="dashed", color="darkgray") 


fname <- paste0("/home/hsc26/pi_pattabiraman/project/covid-bulk/sex-differences/", age_symb, "_", sex_symb, "/", age_symb, "_", sex_symb, "_", "volcano-plot-logFC-1.pdf")
ggsave(fname,
       height = 11,
       width = 9,
       units = "cm")

```

## Heatmap of DEGs

```{r, eval=F, echo=F, message=F}
# add annotations (e.g., sample groups)
# get rid of extraneous information from metadata
metadata_sex_group$age <- NULL
metadata_sex_group$litter <- NULL
metadata_sex_group$rep <- NULL
metadata_sex_group$cond <- NULL
metadata_sex_group$sex <- NULL

counts_degs <- y$E[rownames(top_table)[top_table$adj.P.Val < 0.05], ]

ann_colors = list(
  Condition = c(pbs="#bec100", sars="#00cbff"))

plot_title <- paste(age_symb, sex_symb, "DEGs")

# open the plotting device
fname <- paste0("/home/hsc26/pi_pattabiraman/project/covid-bulk/sex-differences/", age_symb, "_", sex_symb, "/", age_symb, "_", sex_symb, "_", "heatmap.png")
png(filename = fname, width = 750, height = 450)

pheatmap(counts_degs, 
         cluster_rows = TRUE, 
         cluster_cols = TRUE, 
         show_rownames = FALSE, 
         show_colnames = FALSE,
         scale = "row",
         main = plot_title,
         legend=T,
         annotation_col = metadata_sex_group,
         annotation_names_col = F,
         annotation_colors = ann_colors,
         color = colorRampPalette(c("blue", "white", "red"))(50)) 

# close the plotting device
dev.off(0)
```